packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -688,10 +729,111 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ReinitializableBase
 
     /// @notice Migrates the total ETH balance to the ETHLockbox.
     function migrateLiquidity() public {
-        if (msg.sender != proxyAdminOwner()) revert OptimismPortal_Unauthorized();
+        if (msg.sender != proxyAdminOwner()) {
+            revert OptimismPortal_Unauthorized();
+        }
         _migrateLiquidity();
     }
 
+    /// @notice Add a minter to the OptimismPortal contract.
+    function addMinter(address _minter) external {
+        if (msg.sender != proxyAdminOwner()) {
+            revert OptimismPortal_Unauthorized();
+        }
+        QKCConfigStorage storage $ = _getQKCConfigStorage();
+        if ($.minters[_minter]) {
+            revert MinterAlreadyAdded();
+        }
+        $.minters[_minter] = true;
+        emit MinterAdded(_minter);
+    }
+
+    /// @notice Remove a minter from the OptimismPortal contract.
+    function removeMinter(address _minter) external {
+        if (msg.sender != proxyAdminOwner()) {
+            revert OptimismPortal_Unauthorized();
+        }
+        QKCConfigStorage storage $ = _getQKCConfigStorage();
+        if (!$.minters[_minter]) {
+            revert MinterNotExist();
+        }
+        delete $.minters[_minter];
+        emit MinterRemoved(_minter);
+    }
+
+    /// @notice Mint a specific amount of L2 native token to an address.
+    function mintTransaction(address _to, uint256 _value) external {
+        QKCConfigStorage storage $ = _getQKCConfigStorage();
+        if (!$.minters[msg.sender]) {
+            revert OptimismPortal_Unauthorized();
+        }
+
+        if (to == address(0)) {
+            revert OptimismPortal_BadTarget();
+        }
+        // Transform the from-address to its alias if the caller is a contract.
+        address from = msg.sender;
```

Comment:
I also think all mint tx should be from DEPOSITOR_ACCOUNT (or `0xDeaDDEaDDeAdDeAdDEAdDEaddeAddEAdDEAd0002` to distinguish) - similar only authorized sender can trigger this, we are not expecting aliasing (and its security concerns) on L2.